diff --git a/source/data_model/hdf5/CMakeLists.txt b/source/data_model/hdf5/CMakeLists.txt
index 5505da3..f97f939 100644
--- a/source/data_model/hdf5/CMakeLists.txt
+++ b/source/data_model/hdf5/CMakeLists.txt
@@ -37,8 +37,14 @@ target_compile_definitions(lue_hdf5
         H5_USE_110_API
 )
 
+target_compile_definitions(lue_hdf5
+    PUBLIC
+        ${HDF5_C_DEFINITIONS}
+)
+
 target_include_directories(lue_hdf5
     PUBLIC
+        ${HDF5_C_INCLUDE_DIRS}
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
 )
@@ -46,7 +52,7 @@ target_include_directories(lue_hdf5
 target_link_libraries(lue_hdf5
     PUBLIC
         lue::shared
-        HDF5::HDF5
+        ${HDF5_C_LIBRARIES}
     PRIVATE
         Boost::headers
         fmt::fmt
